packages/versioning/src/versioning.ts

```diff
@@ -717,6 +724,8 @@ function cacheVersion(key: Type, versions: [Namespace, VersionMap] | []) {
 }
 
 export function getVersionsForEnum(program: Program, en: Enum): [Namespace, VersionMap] | [] {
+  // FIXME: We should get the versions for the target namespace, not the namespace the enum
+  // is in
   const namespace = en.namespace;
```

Comment:
This seems like it should be the right place to fix this. We should be pulling the version map for the target namespace, not the enum's namespace. However, this is an exported function, so adding a target namespace would at a minimum be a breaking change.

Reply:
Yeah, this function signature is incompatible with this feature .  I did not find any instances of using this other than in our own versioning code, and all of those instances ignore the namespace.  I would agree that we should change this function signature

Reply:
I was able to update this logic and essentially preserve the signature. It uses the reverse lookup table to get the namespaces that version based on the enum rather than the namespace the enum was declared in. Then it uses existing logic.

Reply:
I worry about some instance somewhere depdning on this namespace matching the namespace or parent of some type, but I think this is OK for now. I think we should file an issue to look at usage of this namespace (perhaps it should be a set of namespaces?]